nvfuser/__init__.py

```diff
@@ -202,6 +213,8 @@ def execute(
             self.definition()
             self._finalize_definition()
 
+        self.multidevice_schedule()
```

Comment:
I think `multidevice_schedule` and `schedule` conflict with each other. Should there be an XOR check here?

Reply:
```python
# NOTE: multidevice_schedule and schedule are not defined in the FusionDefinition base class. 

from typing import Callable
has_multidevice_schedule_fn = hasattr(self, 'multidevice_schedule') and isinstance(self.multidevice_schedule, Callable)
has_schedule_fn = hasattr(self, 'schedule') and isinstance(self.schedule, Callable)

# multidevice_schedule is incompatible with schedule, so check that only one is defined.
assert has_multidevice_schedule_fn ^ has_schedule_fn

if has_multidevice_schedule_fn:
  self.multidevice_schedule()

if has_schedule_fn:
   self._setup_schedule(inputs)
   self.schedule()
   self._finalize_schedule(inputs)
```

Reply:
> schedule() should be implemented by child class! 

I've been using decorators to define the schedule function so this ^^^ child class requirement is cumbersome.

Decorator example:
```python
def schedule_decorator(fd):
  def schedule_fn():
    # do some scheduling
  fd.schedule = schedule_fn
  return fd
```

Reply:
I believe it's not XOR but NAND. Anyhow, I think the latest commit has addressed your concern.

Reply:
True. Both functions are optional.